hack/build-and-push.sh

```diff
@@ -133,95 +180,324 @@ if [ "$SKIP_BUILD" == "" ]; then
+inject_bundle_ref_to_pipelines() {
+    local -r task_name=$1
+    local -r task_version=$2
+    local -r task_bundle_with_digest=$3
+    sub_expr_1="
+        (.spec.tasks[].taskRef | select(.name == \"${task_name}\" and .version == \"${task_version}\" ))
+        |= {\"resolver\": \"bundles\", \"params\": [ { \"name\": \"name\", \"value\": \"${task_name}\" } , { \"name\": \"bundle\", \"value\": \"${task_bundle_with_digest}\" }, { \"name\": \"kind\", \"value\": \"task\" }] }
+    "
+    sub_expr_2="
+        (.spec.finally[].taskRef | select(.name == \"${task_name}\" and .version == \"${task_version}\" ))
+        |= {\"resolver\": \"bundles\", \"params\": [ { \"name\": \"name\", \"value\": \"${task_name}\" } , { \"name\": \"bundle\", \"value\": \"${task_bundle_with_digest}\" },{ \"name\": \"kind\", \"value\": \"task\" }] }
+    "
+    for filename in "$GENERATED_PIPELINES_DIR"/*.yaml "$CORE_SERVICES_PIPELINES_DIR"/*.yaml
+    do
+        yq e "$sub_expr_1" -i "${filename}"
+        yq e "$sub_expr_2" -i "${filename}"
+    done
+}
+
+# Get task version from task definition rather than the version in the directory path.
+# Arguments: task_file
+# The version is output to stdout.
+get_concrete_task_version() {
+    local -r task_file=$1
+    # Ensure an empty string is output rather than string "null" if the version label is not present
+    yq '.metadata.labels."app.kubernetes.io/version"' "$task_file" | sed '/null/d' | tr -d '[:space:]'
+}
+
+# Build and push a task as a Tekton bundle, that is tagged a floating tag and a fixed tag.
+# The task bundle reference with digest is output to stdout in the last line,
+# that is extracted from tkn-bundle-push.
+build_push_task() {
+    local -r task_dir=$1
+    local -r prepared_task_file=$2
+    local -r task_bundle=$3
+    local -r task_file_sha=$4
+    # local -r extra_annotations=${5:-""}
+
+    local -r task_description=$(yq e '.spec.description' "$prepared_task_file" | head -n 1)
+
+    local -a ANNOTATIONS=()
+    ANNOTATIONS+=("org.opencontainers.image.source=${VCS_URL}")
+    ANNOTATIONS+=("org.opencontainers.image.revision=${VCS_REF}")
+    ANNOTATIONS+=("org.opencontainers.image.url=${VCS_URL}/tree/${VCS_REF}/${task_dir}")
+    ANNOTATIONS+=("org.opencontainers.image.version=$(get_concrete_task_version "$prepared_task_file")")
+    # yq will return null if the element is missing.
+    if [[ "${task_description}" != "null" ]]; then
+        ANNOTATIONS+=("org.opencontainers.image.description=${task_description}")
+    fi
+    if [ -f "${task_dir}/README.md" ]; then
+        ANNOTATIONS+=("org.opencontainers.image.documentation=${VCS_URL}/tree/${VCS_REF}/${task_dir}README.md")
+    fi
+    if [ -f "${task_dir}/TROUBLESHOOTING.md" ]; then
+        ANNOTATIONS+=("dev.tekton.docs.troubleshooting=${VCS_URL}/tree/${VCS_REF}/${task_dir}TROUBLESHOOTING.md")
+    fi
+    if [ -f "${task_dir}/USAGE.md" ]; then
+        ANNOTATIONS+=("dev.tekton.docs.usage=${VCS_URL}/tree/${VCS_REF}/${task_dir}USAGE.md")
```

Comment:
`${task_dir}README.md` etc.: all of these look like they assume `${task_dir}` always ends with `/`. But I don't think it does